components/experimental/src/units/ratio.rs

```diff
@@ -167,14 +209,21 @@ impl FromStr for IcuRatio {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let mut parts = s.split('/');
-        let num_str = parts.next().unwrap_or("0");
-        let den_str = parts.next().unwrap_or("1");
+
+        let numerator = match parts.next() {
+            Some(num_str) => BigInt::from_str(num_str).map_err(IcuRatioError::BigIntParseError)?,
```

Comment:
nit: you're now parsing the `BigInt`s even if you return `IcuRatioError::InvalidRatioString` later. Defer this.

Reply:
in order to do this, we have two ways:

1- use unwrap_or: let numerator = parts.next().unwrap_or("0")
2- collect in a vec: ``` let parts: Vec<&str> = s.split('/').collect();

        if parts.len() > 2 {
            return Err(IcuRatioError::InvalidRatioString);
        }
```

given that the case of `"1/1/1"` is not common, so, I prefer to use the current approach.

Reply:
3- Use `denominator`/`nominator` variables of type `Option<&str>`

Reply:
done

Reply:
done